src/examples/pongxr/scene.js

```diff
@@ -0,0 +1,200 @@
+/*****
+WHEN CORE IS WORKING:
+- TEST IN THE SOCIAL PLACES
+- RESPONSIVE DESIGN: PHONE
+- RESPONSIVE DESIGN: MAGIC LEAP
+*/
+
+import { Scene, Object3D } from "three";
+import XRInput from "../../engine/xrinput"
+import PeerConnection from '../../engine/networking/PeerConnection'
+import State from "../../engine/state";
+import Physics from "../../engine/physics/physics"
+
+import HostBot from "../../engine/util/hostbot"
+import Level from "./level"
+import Paddle from "./paddle"
+import Ball from "./ball"
+import PlacementCube from "./placementcube"
+
+const scene = new Scene();
+const networking = new PeerConnection(scene);
+const hostBot = new HostBot(networking);
+let ball, placementCube;
+let paddles = [];
+
+// custom States & events
+State.eventHandler.registerEvent('gameover');
+const GameState = { placement: 1, play: 2 }
+
+Physics.enableDebugger(scene);
+
+const createPongLevel = (placementCube) =>
+{
+    const targetPosition = placementCube.position;
+    const targetRotation = placementCube.rotation;
+
+    State.GameState = GameState.play;
```

Comment:
Oh, I see. I like how you're using this. The part of my mind that develops in languages like C# and C++ is trying to understand if this is better than creating a constant variable, something like `currentGameState` since `State.GameState` is only used in this file (as far as I can tell. The part of my mind that develops in languages like JS and Python thinks this is super clear and I like that you're expanding the use of `State` to track local scene game state.

Reply:
you're absolutely right, this is more for the purposes of illustration. 
I'm not demonstrating anywhere else how to use `State` as global app state and while this might be a bit coerced, I think it's illustrative of a particular pattern - registering custom state and event handling at the _app_ level without doing surgery on State.js. 

It's definitely an edge case because I didn't want to expand the XRInput part of `scene.js` to its own file - overmodularization is a slippery slope - but yes, that's the only place currently using that global state. 

Of course, we can add state checks in the various factories (ball, placementCube) to demonstrate the value of global state.

Reply:
Thanks for the explanation. I'm going to borrow this pattern